spec/api-ipc-renderer-spec.js

```diff
@@ -177,6 +177,47 @@ describe('ipc renderer module', () => {
       contents.loadFile(path.join(fixtures, 'pages', 'ping-pong.html'))
     })
 
+    it('sends message to WebContents (context isolation)', done => {
```

Comment:
please use the `generateSpecs` pattern as here [spec/api-browser-window-spec.js](https://github.com/electron/electron/blob/744f737cba67cd6f59ea598aa00795fa4375a929/spec/api-browser-window-spec.js#L1426) to avoid duplicate code

Reply:
also modify the existing test to use the new preload script

Reply:
another example in [spec/api-crash-reporter-spec.js](https://github.com/electron/electron/blob/744f737cba67cd6f59ea598aa00795fa4375a929/spec/api-crash-reporter-spec.js#L191)

Reply:
I've modified the tests for you: https://github.com/electron/electron/commit/8c9589bd9ed3f9e979194670e378579ab158bb86

Reply:
Thanks so much!

Reply:
Sorry, I'm not a frequent GItHub user (Phabricator at work), do I need to update the pull request with your commit?

Reply:
Yes. The other option would be to abandon this PR and create a new one from my branch.

Reply:
Or merge this as is and refactor the tests in a follow up PR.

Reply:
I seem to have mucked things up, so I'm just going to abandon this and create a new pull request from your branch.